From 66bb413b20c3ae24439b6ccbdb846fa80c46de8c Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 29 Nov 2023 23:49:00 +0100
Subject: [PATCH 68/N] fixup! Allow overriding the home directory via the HOME
 variable

In preparation for integrating the patches that made it upstream via
https://inbox.sourceware.org/cygwin-patches/cover.1684753872.git.johannes.schindelin@gmx.de/,
let's revert the original patches from Git for Windows' branch thicket.

This reverts commit 81a0e5d4a9566ebc22b857de5c61f0e12fa32192.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/local_includes/cygheap.h |  3 +-
 winsup/cygwin/uinfo.cc                 | 49 --------------------------
 2 files changed, 1 insertion(+), 51 deletions(-)

diff --git a/winsup/cygwin/local_includes/cygheap.h b/winsup/cygwin/local_includes/cygheap.h
index c33f378..4828e57 100644
--- a/winsup/cygwin/local_includes/cygheap.h
+++ b/winsup/cygwin/local_includes/cygheap.h
@@ -406,8 +406,7 @@ public:
     NSS_SCHEME_UNIX,
     NSS_SCHEME_DESC,
     NSS_SCHEME_PATH,
-    NSS_SCHEME_FREEATTR,
-    NSS_SCHEME_ENV
+    NSS_SCHEME_FREEATTR
   };
   struct nss_scheme_t {
     nss_scheme_method	method;
diff --git a/winsup/cygwin/uinfo.cc b/winsup/cygwin/uinfo.cc
index fefd984..b910878 100644
--- a/winsup/cygwin/uinfo.cc
+++ b/winsup/cygwin/uinfo.cc
@@ -738,8 +738,6 @@ cygheap_pwdgrp::nss_init_line (const char *line)
 		    scheme[idx].method = NSS_SCHEME_UNIX;
 		  else if (NSS_CMP ("desc"))
 		    scheme[idx].method = NSS_SCHEME_DESC;
-		  else if (NSS_CMP ("env"))
-		    scheme[idx].method = NSS_SCHEME_ENV;
 		  else if (NSS_NCMP ("/"))
 		    {
 		      const char *e = c + strcspn (c, " \t");
@@ -928,40 +926,6 @@ fetch_from_path (cyg_ldap *pldap, PUSER_INFO_3 ui, cygpsid &sid, PCWSTR str,
   return ret;
 }
 
-static size_t
-fetch_env(LPCWSTR key, char *buf, size_t size)
-{
-  WCHAR wbuf[32767];
-  DWORD max = sizeof wbuf / sizeof *wbuf;
-  DWORD len = GetEnvironmentVariableW (key, wbuf, max);
-
-  if (!len || len >= max)
-    return 0;
-
-  len = sys_wcstombs (buf, size, wbuf, len);
-  return len && len < size ? len : 0;
-}
-
-static char *
-fetch_home_env (void)
-{
-  char home[32767];
-  size_t max = sizeof home / sizeof *home, len;
-
-  if (fetch_env (L"HOME", home, max)
-      || ((len = fetch_env (L"HOMEDRIVE", home, max))
-        && fetch_env (L"HOMEPATH", home + len, max - len))
-      || fetch_env (L"USERPROFILE", home, max))
-    {
-      tmp_pathbuf tp;
-      cygwin_conv_path (CCP_WIN_A_TO_POSIX | CCP_ABSOLUTE,
-	  home, tp.c_get(), NT_MAX_PATH);
-      return strdup(tp.c_get());
-    }
-
-  return NULL;
-}
-
 char *
 cygheap_pwdgrp::get_home (cyg_ldap *pldap, cygpsid &sid, PCWSTR dom,
 			  PCWSTR dnsdomain, PCWSTR name, bool full_qualified)
@@ -1021,10 +985,6 @@ cygheap_pwdgrp::get_home (cyg_ldap *pldap, cygpsid &sid, PCWSTR dom,
 		}
 	    }
 	  break;
-	case NSS_SCHEME_ENV:
-	  if (RtlEqualSid (sid, cygheap->user.sid ()))
-	    home = fetch_home_env ();
-	  break;
 	}
     }
   return home;
@@ -1059,10 +1019,6 @@ cygheap_pwdgrp::get_home (PUSER_INFO_3 ui, cygpsid &sid, PCWSTR dom,
 	  home = fetch_from_path (NULL, ui, sid, home_scheme[idx].attrib,
 				  dom, NULL, name, full_qualified);
 	  break;
-	case NSS_SCHEME_ENV:
-	  if (RtlEqualSid (sid, cygheap->user.sid ()))
-	    home = fetch_home_env ();
-	  break;
 	}
     }
   return home;
@@ -1082,7 +1038,6 @@ cygheap_pwdgrp::get_shell (cyg_ldap *pldap, cygpsid &sid, PCWSTR dom,
 	case NSS_SCHEME_FALLBACK:
 	  return NULL;
 	case NSS_SCHEME_WINDOWS:
-	case NSS_SCHEME_ENV:
 	  break;
 	case NSS_SCHEME_CYGWIN:
 	  if (pldap->fetch_ad_account (sid, false, dnsdomain))
@@ -1147,7 +1102,6 @@ cygheap_pwdgrp::get_shell (PUSER_INFO_3 ui, cygpsid &sid, PCWSTR dom,
 	case NSS_SCHEME_CYGWIN:
 	case NSS_SCHEME_UNIX:
 	case NSS_SCHEME_FREEATTR:
-	case NSS_SCHEME_ENV:
 	  break;
 	case NSS_SCHEME_DESC:
 	  if (ui)
@@ -1229,8 +1183,6 @@ cygheap_pwdgrp::get_gecos (cyg_ldap *pldap, cygpsid &sid, PCWSTR dom,
 		sys_wcstombs_alloc (&gecos, HEAP_NOTHEAP, val);
 	    }
 	  break;
-	case NSS_SCHEME_ENV:
-	  break;
 	}
     }
   if (gecos)
@@ -1257,7 +1209,6 @@ cygheap_pwdgrp::get_gecos (PUSER_INFO_3 ui, cygpsid &sid, PCWSTR dom,
 	case NSS_SCHEME_CYGWIN:
 	case NSS_SCHEME_UNIX:
 	case NSS_SCHEME_FREEATTR:
-	case NSS_SCHEME_ENV:
 	  break;
 	case NSS_SCHEME_DESC:
 	  if (ui)
